### PR TITLE
Implementing Theme switching for Fluent

### PR DESCRIPTION
**Pull request by dipeshmsft**

### CODE REVIEW

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager.cs [hunk 1]
```diff
@@ -260,8 +262,10 @@ private static void FindFluentThemeAndColorDictionary(out ResourceDictionary flu
 
     private static readonly string _regPersonalizeKeyPath = "HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize";
 
-    private static readonly string fluentResourceDictionaryUri = "pack://application:,,,/PresentationFramework.Fluent;component/Resources/Fluent.xaml";
-    private static readonly string fluentColorResourceUriPart = "pack://application:,,,/PresentationFramework.Fluent;component/Resources/Theme/";
+    // private static readonly string fluentResourceDictionaryUri = "pack://application:,,,/PresentationFramework.Fluent;component/Resources/Fluent.xaml";
+    // private static readonly string fluentColorResourceUriPart = "pack://application:,,,/PresentationFramework.Fluent;component/Resources/Theme/";
+
+    private static readonly string fluentThemeResoruceDictionaryUri = "pack://application:,,,/PresentationFramework.Fluent;component/Themes/";
```
Comment (miloush) [inline src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager.cs:268]: typo resoruce

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager.cs [hunk 2]
```diff
@@ -215,43 +223,37 @@ private static void AddOrUpdateThemeResources(Uri dictionaryUri)
...
         foreach (ResourceDictionary mergedDictionary in Application.Current.Resources.MergedDictionaries)
         {
             if (mergedDictionary.Source != null)
             {
-                if (mergedDictionary.Source.ToString() == fluentResourceDictionaryUri)
-                {
-                    fluentThemeDictionary = mergedDictionary;
-                }
-                else if (mergedDictionary.Source.ToString().StartsWith(fluentColorResourceUriPart))
+                if (mergedDictionary.Source.ToString().Contains(fluentThemeResoruceDictionaryUri))
```
Comment (miloush) [inline src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager.cs:248]: isn't the only possible case caught by StartsWith?

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager3.cs [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+using System.Windows.Media;
+using System.Windows.Appearance;
+
+namespace System.Windows;
+
+public static class ThemeManager3
+{
+    static ThemeManager3()
+    {
+        if(Application.Current != null)
```
Comment (miloush) [inline src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager3.cs]: could do Application.Current?.Theme == "None"

(btw this whole file seems to have weird code style with no space before opening parenthesis

Comment (dipeshmsft) [reply src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager3.cs]: Yeah just updated that. Btw take a look at ThemeManager3 only. I am removing ThemeManager and replacing it with ThemeManager3 ( I will rename this in main PR )

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager3.cs [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+        // }
+    }
+
+    internal static void OnApplicationThemeChanged(string oldTheme, string newTheme)
+    {
+        IgnoreAppResourceDictionaryChanges = true;
+
+        if(!IsFluentThemeEnabled)
+        {
+            IgnoreAppResourceDictionaryChanges = false;
```
Comment (miloush) [inline src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager3.cs]: it's safer to do
```C#
try { /* the whole method */ }
finally { IgnoreAppResourceDictionaryChanges = false; }
```

**Pull request closed without merging**